app/api/v2/identity/users.rb

```diff
@@ -60,9 +60,10 @@ def parse_refid!
           activity_record(user: user.id, action: 'signup', result: 'succeed', topic: 'account')
 
           publish_confirmation(user, language, Barong::App.config.domain)
-          open_session(user)
+          token = SecureRandom.hex(10)
```

Comment:
this must be in open_session, to me it's part of this function

Reply:
I need to have it to give to the entity as a param.
If I will move it in the function, then I will need function to return value. (Which is already misunderstanding of open_session aim)
Save value in a var and give to entity .

It will become ugly